### PR TITLE
Support for prefix/suffix based input file hunting.

### PR DESCRIPTION
**Pull request by asheshambasta**
Adds support for prefix or suffix based input files at standard locations (linux support only). 

See the commits for a detailed implementation, and commit messages for summaries. 

### Pending 

- [ ] Changes in docs
  @david-janssen if you have ideas on what should be in the docs, let me know.

### CODE REVIEW

File: .dir-locals.el [hunk 1]
```diff
@@ -0,0 +1,7 @@
+(
+ (nil . (
+         ;; (haskell-mode-stylish-haskell-path . "stylish-haskell")
```
Comment (asheshambasta) [inline .dir-locals.el:3]: This is pending a decision on formatters, see https://github.com/kmonad/kmonad/issues/266#issuecomment-846571515

**Comment by david-janssen** [comment]
Hey Ashesh,

Thank you for rebasing this on keycode-refactor. I like it a lot. Looking at it a bit closer, there is only 1 change I'd like to ask you to make. You've currently written it so that we end up doing IO in the 'TokenJoiner', but I'd really like to avoid that. Ideally, the 'TokenJoiner' (and all of the parsing code) just generates pure tokens from a config-file, regardless of the platform or filesystem on which they are run.

I'd like to move the IO parts you've inserted into the joiner into `withKeyInput` in 'KMonad.App.KeyIO.Linux' instead. Once that is done, I'll be happy to merge. We still have to figure out where we want to put the docs, but I think improving the docs is a general issue that we have to come up with a good solution for first anyways.

**Comment by asheshambasta** [comment]
@david-janssen Hi, have you had the chance to look at this?

**Comment by david-janssen** [comment]
Hey @asheshambasta, sorry for being away for a bit, got ill again. I submitted some changes to your branch here: https://github.com/asheshambasta/kmonad/pull/2, and was waiting for you to respond there. I wasn't entirely sure how I might submit my own direct changes to your pull-request. Is there some better way?

**Comment by 6gk** [comment]
@david-janssen If the submitter of the PR hasn't unchecked the option (it's on by default), maintainers of the original repository can push directly to the given branch.

image

**Comment by david-janssen** [comment]
I think I tried that and it didn't work. Maybe I did it wrong, or maybe Ashesh disabled the feature. Let's see how he replies.

**Pull request closed without merging**